.github/workflows/tests.yaml

```diff
@@ -0,0 +1,47 @@
+name: tests
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+      - uses: actions/checkout@v4
+        
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
```

Comment:
I've never heard of `snok` - what is that? Why not simply `pipx install poetry`?

Reply:
Done. From prosnet workflows I deduce that `pipx` is available either on the `ubuntu-latest` image or from the `setup-python` action, so apparantly there is no need to install it separately.